### PR TITLE
Stubbed project for 2023 robot 

### PR DESCRIPTION
**Pull request by mmcadamsms**
Project named smokey for now.  Project uses WPILib 2023 ... all contributors need to upgrade to latest tools!

### CODE REVIEW

File: smokey/src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    // standard stuff constants - motors rotation, etc.
+    public static final double DegreesPerRevolution = 360.0;
+    // NEO maximum RPM 
+    public static final double neoMaximumRevolutionsPerMinute = 5676;
+    // this uses the halls effect sensor when plugged into the spark max
+    // see: https://www.revrobotics.com/rev-21-1650/ where it says "42 counts per rev."
+    public static final double RevNeoEncoderTicksPerRevolution = 42;
+
+    // *****************************************************************
+    // ball handler system constants
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/Constants.java:63]: should we remove the ball handler constants? If you meant to keep them as a starter for the grabber, then should we rename them?

File: smokey/src/main/java/frc/robot/Constants.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    public static final int PneumaticsControlModuleForwardChannel = 0;
+    public static final int PneumaticsControlModuleReverseChannel = 1;
+
+    // *****************************************************************
+    // input device constants
+    public static final int portDriverController = 1;
+    public static final int portCoDriverController = 2;
+
+    // *****************************************************************
+    // telescoping arm system constants
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/Constants.java:86]: should we remove the telescoping arm constants?  Or did you mean to keep them for the arm that will go on smokey?

File: smokey/src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,201 @@
+// ************************************************************
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java:1]: this isn't a copy of the latest from Veer.  It seems like the vanilla one from swerve drive specialties.  Is there a reason not to start from the latest on Veer and modify from there?

File: smokey/src/main/java/frc/robot/commands/DefaultDriveCommand.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
+// ************************************************************
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/commands/DefaultDriveCommand.java:1]: this isn't the latest from Veer. Is there a reason not to start from the latest on Veer and modify from there?

**Review by josieammer — APPROVED** [review]
The drivetrain subsystem and default drive command doesn't seem to be the latest from veer. 
Some vestiges of ball handler constants still in Constants.java.  Otherwise looks good.  Marking approved to allow you to merge this and then fix in the next commit, or fix now before committing.

File: smokey/src/main/java/frc/robot/Constants.java (hunk 1 above)
Comment (mmcadamsms) [reply smokey/src/main/java/frc/robot/Constants.java]: agree ... done

File: smokey/src/main/java/frc/robot/Constants.java (hunk 2 above)
Comment (mmcadamsms) [reply smokey/src/main/java/frc/robot/Constants.java]: agree ... done

File: smokey/src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java (hunk 3 above)
Comment (mmcadamsms) [reply smokey/src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java:1]: so this is a copy of what is in the main branch ... do you know the branch I should get it out of?

File: smokey/src/main/java/frc/robot/commands/DefaultDriveCommand.java (hunk 4 above)
Comment (mmcadamsms) [reply smokey/src/main/java/frc/robot/commands/DefaultDriveCommand.java:1]: again ... I got this from Veer in main branch ... the other code must be off in some other feature branch

Comment (mmcadamsms) [reply smokey/src/main/java/frc/robot/commands/DefaultDriveCommand.java:1]: Found some stuff out in the swerve-shuffleboard-to-navX branch that hopefully is closer.  We need to have folks finish their work and get it to main going forward.

Comment (josieammer) [reply smokey/src/main/java/frc/robot/commands/DefaultDriveCommand.java:1]: What's on the main branch of veer looks right
https://github.com/Team4682CyBears/ChargedUp2023/blob/main/veer/src/main/java/frc/robot/commands/DefaultDriveCommand.java

But that's not what this file has.  For example, line 22 should have a boolean flag to toggle field oriented drive... 
https://github.com/Team4682CyBears/ChargedUp2023/blob/5a3f202910f254a7ca19229bdff7759d9ae728b8/veer/src/main/java/frc/robot/commands/DefaultDriveCommand.java#L22

**Pull request merged**